### DISCUSSION TITLE
ChatGPT - Current state for .NET - and in general

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#31] [General]
# ChatGPT - Current state for .NET - and in general

ChatGPT is going viral right now. Besides funny conversations, you can use that tool to generate also code for you.
But does it really hold up to its reputation and it is a serious pairing partner? Let's see.

https://steven-giesel.com/blogPost/566d2090-eb8e-492d-ac99-fb6f14b6e042

**Comment by 274188A** [comment]
The concept of the AI retaining 'context' is super important - even with the password example, if used will "warn" you it is super weak.  It seems a little disingenuous to leave that out of your article.  I could subsequently ask for a stronger encryption library - with salting etc and it would oblige.

Comment produced below:
"It is important to note that this is just an example and should not be used in production as-is, as it does not include proper salt handling or iteration count. It is recommended to use a well-tested library such as ASP.NET Core's built-in password hashing utilities for hashing passwords in a production application."

**Reply by linkdotnet** [reply]
Hmm I tried it again and ChatGPT isn't mentioning any of this. Instead, I got this:
> In this example, we are using the SHA-256 algorithm to compute the hash of the password. You can choose to use a different algorithm by replacing SHA256.Create() with the appropriate constructor for the desired algorithm.
> 
> It's important to note that the resulting hash is not the same as the original password, and it cannot be used to retrieve the password. Instead, the hash can be used to verify that a given password is correct by computing the hash of the given password and comparing it to the original hash.
> 
> Hope this helps! Let me know if you have any other questions.
